### DISCUSSION TITLE
How do get estimated gas fees using usePrepareContractWrite

### DISCUSSION THREAD

**Discussion by martinang2** [wevm/wagmi#1602] [Q&A]
Hi Everyone, I'm using the usePrepareContractWrite hook. 

Does anyone know the correct way to extract the estimated gas fees for my transaction?

I've been using the gasLimit from the returned object  `config?.request?.gasLimit?.toBigInt()` however, this number differs greatly from the gas estimation from MetaMask

**Comment by StKostyantin** [comment]
gas price(in gwei) = gas limit * (base fee + priority fee)
